ods_ci/tests/Resources/Files/llm/vllm/download_model.yaml

```diff
@@ -0,0 +1,74 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: vllm-gpt2
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: vlmm-gpt2-claim
+  namespace: vllm-gpt2
+spec:
+  accessModes:
+    - ReadWriteOnce
+  volumeMode: Filesystem
+  resources:
+    requests:
+      storage: 10Gi
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: setup-gpt2-binary
+  namespace: vllm-gpt2
+  labels:
+    gpt-download-pod: 'true'
+spec:
+  volumes:
+    - name: model-volume
+      persistentVolumeClaim:
+        claimName: vlmm-gpt2-claim
+  restartPolicy: Never
+  initContainers:
+    - name: fix-volume-permissions
+      image: quay.io/quay/busybox:latest
+      imagePullPolicy: IfNotPresent
+      securityContext:
+        allowPrivilegeEscalation: true
+      resources:
+        requests:
+          memory: "64Mi"
+          cpu: "250m"
+          nvidia.com/gpu: "1"
+        limits:
+          memory: "128Mi"
+          cpu: "500m"
+          nvidia.com/gpu: "1"
+      command: ["sh"]
+      args: ["-c", "chown -R 1001:1001 /mnt/models"]
+      volumeMounts: 
+        - mountPath: "/mnt/models/"
+          name: model-volume
+  containers:
+    - name: download-model
+      image: registry.access.redhat.com/ubi9/python-311:latest
+      imagePullPolicy: IfNotPresent
+      securityContext:
+        allowPrivilegeEscalation: true
+      resources:
+        requests:
+          memory: "1Gi"
+          cpu: "1"
+          nvidia.com/gpu: "1"
+        limits:
+          memory: "1Gi"
+          cpu: "1"
+          nvidia.com/gpu: "1"
+      command: ["sh"]
+      args: [ "-c", "pip install --upgrade pip && pip install --upgrade huggingface_hub && python3 -c 'from huggingface_hub import snapshot_download\nsnapshot_download(repo_id=\"gpt2\", local_dir=\"/mnt/models/gpt2\", local_dir_use_symlinks=False)'"]
```

Comment:
we should used the exting method to download the model in PVC since gpt model is already present in our bucket

Reply:
@bdattoma also suggested the same. I'm not sure it would make any difference though?

Reply:
reducing code duplication mainly

Reply:
for disconnected cluster to download model from huggingface directly will be bot possible

Reply:
Agreed we should move to our usual download keyword, but it does not appear to be a trivial change for now. The model in our bucket seems to have multiple formats compared to the one downloaded straight from HF, so we need to determine which one should be used. Reusing the keyword for the SR deploy also does not seem trivial, as the current SR is hardcoded to work with the model downloaded from HF. Will need some time spent reworking this specific aspect in another PR.